### PR TITLE
allow fixed offers to be loaded in tools/dataProcessor.js

### PR DESCRIPTION
**Pull request by hongalex**
First open source pull request that addresses issue #11

Added a file named `fixedInputData.json` in `/data/cashback/` for future developers to specify additional fixed cashback offers that might come up. This still has to follow the schema as defined in the cashback schema file. Currently only has Amazon available, since that was mentioned specifically in the issue. 

One thing I chose to ignore was that due to the way dataProcessor.js handles google search queueing, a search will be performed for user-defined cashback offers as well. I could modify this with a moderate refactor of how searching/search callbacks are handled, but since this file isn't likely to be very large, I left it as is. 

Let me know if anything needs work :)

### CODE REVIEW

File: tools/dataProcessor.js [hunk 1]
```diff
@@ -63,6 +64,14 @@ if (argv.deal) {
  */
 function parseCashbacks() {
 
+  // First, load any predefined fixed cashback offers (e.g. Amazon cashback)
+  let fixedCashbackData = fs.readFileSync(CASHBACK_INPUT_FILE, 'utf8');
```
[the remark this answers is not in the document]
Comment (nareddyt) [reply tools/dataProcessor.js]: Ignore this

**Comment by nareddyt** [comment]
Closes #11 

This is great! Since we only need to show amazon.com for now, refactoring the google searching seems like an overkill. Let's leave it like this for now.

I'll have to test this locally just to be 100% sure the changes work. Once that's done, I'll merge it in.

**Comment by nareddyt** [comment]
**LGTM**, will merge now 👍 

Here is proof that this adds the amazon offer when visiting www.amazon.com. The extension on the left (with the colored icon) is my local install, the one of the right right (with the gray icon) is the production one. As seen, the local install displayed the amazon.com cashback offer.

image

Since this is a data-related changed, I had to switch a flag in `src/background.js` to read the data from a local directory instead of using the hosted production data. This flag can be [found here](https://github.com/nareddyt/discover-rewards-notifier/blob/master/src/background.js#L9).

**Comment by nareddyt** [comment]
Updated cashback data for www.amazon.com is now live! The hosted json can [be viewed here](https://www.tejunareddy.com/discover-rewards-notifier/cashbacks.json).

All users will receive this data automatically. The chrome extension is designed to download the hosted json every hour, so there is no need to make a new release. Within the next hour, all users should be seeing this cashback offer on www.amazon.com! Congrats on your first change :)

**Comment by hongalex** [comment]
Cool, thanks for the help! I'll leave some of the easier issues for future first-timers to pick up, might look into some of the harder ones once I get the chance. Thanks again!

**Pull request merged**